lib/matplotlib/animation.py

```diff
@@ -193,6 +193,9 @@ def setup(self, fig, outfile, dpi=None):
             The DPI (or resolution) for the file.  This controls the size
             in pixels of the resulting movie file.
         """
+        outdirectory = Path(outfile).parent
+        if not outdirectory.exists():
+            raise ValueError(f"{outdirectory} does not exist!")
```

Comment:
(Usually) no final punctuation in error messages (https://github.com/matplotlib/matplotlib/pull/20211 and IIRC @timhoffm made a comment to the same effect in a recent PR review).